contracts/liquidity_hub/fee_collector/src/queries.rs

```diff
@@ -12,3 +17,84 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
     let config = CONFIG.load(deps.storage)?;
     Ok(config)
 }
+
+pub fn query_accrued_fees(deps: Deps, collect_fees_for: CollectFeesFor) -> StdResult<Vec<Asset>> {
+    let mut query_fees_messages: Vec<Asset> = Vec::new();
+
+    match collect_fees_for {
+        CollectFeesFor::Contracts { contracts } => {
+            for contract in contracts {
+                let response = query_fees_for_vault(&deps, contract)?;
```

Comment:
I believe it's not possible to query accrued fees for contracts, as this allows you to put in any contract, namely pools and/or vaults mixed together, you won't know what it is. Thus, using `using_query_for_vaults()` in here will fail if you pass pools instead. I would say we return an error if you do the query using `CollectFeesFor::Contracts` with a meaningful message.

Reply:
Thanks, I forgot about it. I added there the error message, but please feel free to edit it in the name of better developer experience :)

Reply:
Also I was thinking, if the fee query for a specific contract might be useful I can add there custom msg enum. Slightly edited `CollectFeesFor`, where would be the option to include also contract type.

Reply:
Well done! To make the error even better, consider defining it as a `ContractError`.

Reply:
Good idea with the `CollectFeesFor` enum. However it would be semantically wrong to use `CollectFeesFor` for queries if we modify the `contracts` option to allow passing an Option for contract type, namely vault or pool.

I woulkd say we do it, but instead of using `CollectFeesFor` in `QueryMsg::AccruedFees`, create and use instead something called `QueryFeesFor` which looks something like this:

```rust
pub enum QueryFeesFor {
    Contracts { contracts: Vec<Contract> },
    Factory {
        factory_addr: String,
        factory_type: FactoryType,
    },
}

pub struct Contract {
    pub address: String,
    pub type: ContractType
}

pub enum ContractType {
    Vault { },
    Pool { },
}

```

By doing so we don't need to define any error as I mentioned before. So far what you are doing is looking good, thanks!